pyvrp/ProblemData.pyi

```diff
@@ -28,16 +90,128 @@ class ProblemData:
-    def depot(self) -> Client: ...
-    def dist(self, first: int, second: int) -> int: ...
-    def distance_matrix(self, *args, **kwargs) -> Any: ...
+    ):
+        """
+        Creates a problem data instance. This instance contains all information
+        needed to solve the vehicle routing problem.
+
+        Parameters
+        ----------
+        coords
+            Array of (x, y) coordinates. The first coordinate at index 0 is
+            assumed to be the depot.
+        demands
+            Array of client demands. The demand at index 0 is assumed to be the
+            depot's demand, and should be zero.
+        nb_vehicles
+            The number of vehicles in this problem instance.
+        vehicle_cap
+            Homogenous vehicle capacity for all vehicles in the problem
+            instance.
+        time_windows
+            Array of (early, late) time windows. The time window at index 0 is
+            assumed to be the depot's time window, and describes the overall
+            time horizon.
+        service_durations
+            Array of service durations, that is, the length of time needed to
+            service a customer upon visiting. The service duration at index 0
+            is assumed to be the depot's service time, and should be zero.
+        duration_matrix
+            A matrix that gives the travel times between clients (and the
+            depot at index 0). Does not have to be symmetric.
+        release_times
+            Release times give the earliest time at which a client's request
+            may leave the depot.
+
+        Notes
+        -----
+        All array data assume that the data at or involving index 0 relates to
+        the depot, and all other indices specify client information.
+        """
+    def client(self, client: int) -> Client:
+        """
+        Returns client data for the given client.
+
+        Parameters
+        ----------
+        client
+            Client number whose information to retrieve.
+
+        Returns
+        -------
+        Client
+            A simple data object containing the requested client's information.
+        """
+    def depot(self) -> Client:
+        """
+        Returns 'client' information for the depot, which is stored internally
+        as the client with number `0`.
+
+        Returns
+        -------
+        Client
+            A simple data object containing the depot's information.
+        """
+    def dist(self, first: int, second: int) -> int:
+        """
+        Returns the travel duration between the first and second argument,
+        according to this instance's travel duration matrix.
+
+        Parameters
+        ----------
+        first
+            Client or depot number.
+        second
+            Client or depot number.
+
+        Returns
+        -------
+        int
+            Travel duration between the given clients.
+        """
+    def distance_matrix(self) -> Any:
```

Comment:
Question: why is this typed `Any` instead of an array or lists of lists?

Reply:
Because this will very soon become a `nd.ndarray` but it isn't yet, and I didn't want to bother typing this yet :)